### DISCUSSION TITLE
Can we make Enumerable<T> readonly?

### DISCUSSION THREAD

**Discussion by khokm** [prisma/prisma#17451] [Q&A]
Hi and thx for greate library.
I try to make query using Enumerable, but i prefer to use immutable types (i.e. `data: ReadonlyArray<string>` ).
So when i pass it to query, i need to cast it to mutable type: `data as string[]`.
I don't think replacing `Array` with `ReadonlyArray` will break anyhing.

https://github.com/prisma/prisma/blob/7f20fe059925d006e07112995fc6b928c4a14d27/packages/client/src/generation/TSClient/common.ts#L321

**Comment by nurul3101** [comment]
Hey @khokm 👋 

Thanks for your kind words and the recommendation.

As this is a change in the codebase, we would need to ensure that all the [Ecosystem Tests](https://github.com/prisma/ecosystem-tests) pass and our engineering team approves of your changes, and they don't foresee this as a breaking change.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
